src/odemis/odemisd/test/modelgen_test.py

```diff
@@ -0,0 +1,165 @@
+option) any later version.
+
+Delmic Acquisition Software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+more details.
+
+You should have received a copy of the GNU General Public License along with Delmic Acquisition Software. If not, see
+http://www.gnu.org/licenses/.
+"""
+import unittest
+import yaml
+
+from odemis.odemisd.modelgen import SafeLoader, ParseError
+
+
+class SafeLoader_extensions_test(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        with open("yaml-merger-combination-include-extend-expected-result.yaml", "r") as f:
+            cls.expected_full_result = yaml.load(f, SafeLoader)
+
+    def test_include(self):
+        """
+        Basic test for the include keyword in a yaml file to include the content of a file as the value of a key.
+        """
+        # __init__ of the CL-detector
+        with open("yaml-merger-include-complete-init-CL-Detector-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"]
+        self.assertEqual(expected_result, data_found)
+
+        # Full CL-detector component
+        with open("yaml-merger-include-full-CL-Detector-component-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = {"CL Detector": self.expected_full_result["CL Detector"]}
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ contains a setting which is overwritten of the CL-detector
+        with open("yaml-merger-include-complete-overwrite-init-CL-Detector-test.odm.yaml", "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"].copy()  # Copy because a value is adjusted
+        expected_result["settle_time"] = ["overwritten", "values", "are", "correctly", "stored"]
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ of the CL-detector using a relative reference in the !include
+        with open(
+            "yaml-merger-relative-path-test/yaml-merger-include-complete-init-relative-path-CL-Detector-test.odm.yaml",
+                "r") as f:
+            data_found = yaml.load(f, SafeLoader)
+
+        # Compare with expected results
+        expected_result = self.expected_full_result["CL Detector"]["init"]
+        self.assertEqual(expected_result, data_found)
+
+        # __init__ of the CL-detector using an absolute reference in the !include
+        with open(
+            "yaml-merger-relative-path-test/yaml-merger-include-complete-init-relative-path-CL-Detector-test.odm.yaml",
```

Comment:
What's the difference between this line and line 67?

Reply:
You could consider adding a `path = "yaml-mer......" ` variable to make this a bit more readable.